bot/cogs/help.py

```diff
@@ -0,0 +1,239 @@
+"""Help system.
+
+This help is nearly perfect, with only one warning (access to a protected
+class) and two info (not enough methods at line 16 and too many arguments at
+line 19), under Flake8, Mypy, Pydocstyle and Pylint.
+"""
+
+from inspect import Parameter
+import textwrap
+import typing as t
+
+import discord
+from discord.ext import commands, menus
+import discord.utils
+
+
+class HelpSource(menus.ListPageSource):
+    """The Help menu."""
+
+    def __init__(
+            self,
+            signature: t.Callable,
+            filter_commands: t.Coroutine,
+            prefix: str,
+            author: discord.User,
+            cogs: dict
+            ) -> None:
```

Comment:
This is what I suggested before, this bracket should be on the same level as the function definition, not with the arguments. 
Also, the arguments should be indented by exactly 4 spaces (1 indent size) not 8 as you're doing here.
This is also the cause of `flake8` error `E123` also detected by Azure Pipelines.
```suggestion
    def __init__(
        self,
        signature: t.Callable,
        filter_commands: t.Coroutine,
        prefix: str,
        author: discord.User,
        cogs: dict
    ) -> None:
```

Reply:
I just don't like having the last line indented less than the next one. Feels wrong

Reply:
you need to though, it's a convention which makes it more readable

Reply:
Anyway, I got many things to export from main to core/bot before doing that

Reply:
As daneel said here, we require this, it is the syntax we use everywhere in our codebase and there is no reason to make this an exception

Reply:
C0330 : wrong hanging indentation before block (add four spaces)
So 8 is fine (else it would be on the same level as the body of the function)